gui.py

```diff
@@ -0,0 +1,189 @@
+        :return: None
+        """
+        self.hex_outer_radius =\
+            min(self.size().height() / (4 * map_size), self.size().height() / (4 * map_size))
+
+    def hex_to_pixel(self, cell: Coordinates) -> tuple[int, int]:
+        """
+        Calculates axis shift in pixels of each Hex Widget dependent on
+        coordinates relative to central Hex Widget
+        :param cell: Coordinates obj
+        :return: pixels shift (x, y)
+        """
+        shift_x = self.hex_outer_radius * (3 / 2 * cell[0])
+        shift_y = self.hex_outer_radius * (math.sqrt(3) / 2 * cell[0] + math.sqrt(3) * cell[1])
+        return shift_x, shift_y
+
+    def refresh_map(self, map_: GameMap, state: GameState) -> None:
+        """
+        Slot connected to presenter signal game_state_updated.
+        Updates widgets in main window dependent on current game content.
+        Create child Hex Widgets of proper type, move Widgets to correct pixel.
+        Every turn sets parent as None for widgets created on previous turn,
+        that decrease reference count of each widget to 0
+        :param map_: GameMap obj
+        :param state: GameState obj
+        :return: None
+        """
+        if self.hex_outer_radius is None:
+            self.set_hex_radius(map_.size)
+        for cell in map_.cells:
+            existing_cell = self.findChild(Hex, f"{cell}")
+            if existing_cell:
+                existing_cell.setParent(None)
+            text = ""
+            color = ucf.HEX_DEFAULT_FILL
+            if cell in map_.obstacles:
+                color = ucf.OBSTACLE_COLOR
+            elif cell in state.get_our_tanks_cells():
+                color = ucf.OUR_TANKS_COLOR
+                tank_id = state.get_our_tank_id(cell)
+                text = state.our_tanks[tank_id].health
+            elif cell in state.enemy_tanks:
+                color = ucf.ENEMY_COLOR
+                text = state.enemy_tanks[cell]
+            elif cell in map_.base:
+                color = ucf.BASE_COLOR
+            elif cell in map_.spawn_points:
+                color = ucf.SPAWN_COLOR
+            hex_ = Hex(self.hex_outer_radius, color, text)
+            hex_.setParent(self)
+            hex_.setObjectName(f"{cell}")
+            shift_x, shift_y = self.hex_to_pixel(cell)
+            hex_.move(self.get_mid_map()[0] + shift_x, self.get_mid_map()[1] + shift_y)
+            hex_.show()
+
+    def show_message(self, text: str) -> None:
+        """
+        Create pop-up message window
+        :param text: text to display
+        :return: None
+        """
+        message = QtWidgets.QMessageBox(text=text, parent=self)
+        message.show()
+
+    def closeEvent(self, event: QtGui.QCloseEvent) -> None:
+        """
+        Close event handler
+        :param event: system generated event
+        :return: None
+        """
+        self.presenter_thread.exit()
+
+
+class Hex(QtWidgets.QWidget):
```

Comment:
I would recommend to think of splitting this module up to separate modules per GUI component.